bandit/plugins/django_username_comparison.py

```diff
@@ -0,0 +1,62 @@
+import bandit
+from bandit.core import cwemap
+from bandit.core import test_properties as test
+
+
+def _check_username_comparison(context) -> bool:
+    if context.node.left.attr != "username":
+        return False
+
+    parent = context.node.left.value or None
+    if hasattr(parent, "attr") and parent.attr != "user":
+        return False
```

Comment:
We may want to avoid looking up for the same attribute twice in a row. Two possible ways exist for that:
```suggestion
    if getattr(parent, "attr", None) != "user":
        return False
```
or
```suggestion
    try:
        if parent.attr != "user":
            return False
    except AttributeError:
        pass
```

We should go for the latter if (and only if) `parent` has an `attr` attribute "most of the time".

Reply:
I had an answer regarding `hasattr` and `getattr` having better performances than `try/except` but it was opinionated. Apparently, [this answer](https://stackoverflow.com/a/903238) supports this point so I'd rather be in favor of `getattr`, let alone for the readability also (fewer lines to express our final intent of just getting an attribute).